src/Nethermind/Nethermind.Optimism/OptimismTxValidator.cs

```diff
@@ -8,21 +8,11 @@
 
 namespace Nethermind.Optimism;
 
-public class OptimismTxValidator : ITxValidator
+public sealed class OptimismTxValidator : ITxValidator
```

Comment:
The existing `OptimismTxValidator` always validates transactions that are of `TxType.Deposit` without performing any validation at all. This does not look correct since there are validations that are performed on all existing `TxType`s like `IntrinsicGasTxValidator` but I might be missing some context.

Reply:
maybe @jmederosalvarado knows more about this?

Reply:
Deposit transactions are special types, I think there should only be one per block on first step. Ping @jmederosalvarado , @deffrian or @flcl42 about it.

Reply:
DepositTxs are mostly validated on L1 smart contract, and later by the CL client. Whatever DepositTx the CL feeds to the EL client is already valid.